x/projects/keeper/project.go

```diff
@@ -62,9 +63,20 @@ func (k Keeper) AddKeysToProject(ctx sdk.Context, projectID string, adminKey str
 	return k.projectsFS.AppendEntry(ctx, projectID, uint64(ctx.BlockHeight()), &project)
 }
 
-func (k Keeper) ChargeComputeUnitsToProject(ctx sdk.Context, project types.Project, cu uint64) (err error) {
-	project.UsedCu += cu
-	return k.projectsFS.ModifyEntry(ctx, project.Index, uint64(ctx.BlockHeight()), &project)
+func (k Keeper) ChargeComputeUnitsToProject(ctx sdk.Context, project types.Project, blockHeight uint64, cu uint64) (err error) {
+	blocks := k.projectsFS.GetEntryVersionsRange(ctx, project.Index, blockHeight, uint64(commontypes.STALE_ENTRY_TIME))
+
+	for _, block := range blocks {
+		var proj types.Project
+		k.projectsFS.ReadEntry(ctx, project.Index, block, &proj)
+		if proj.Snapshot != project.Snapshot {
```

Comment:
what if you have snapshots of 1,2,3 and you need to charge snapshot 2?
this means you will break at one before you even reach 2

Reply:
That cannot happen: snapshot happens once a month, and should not be kept - for the purpose of accepting payments - for as long.  But even if they stay around, this behavior is still correct: a payment to a project version should propagate forward until the next recharge (= snapshot) of that project, and not beyond/into the new period of accounting.